workflow/controller/operator_test.go

```diff
@@ -4021,6 +4021,54 @@ status:
 	}
 }
 
+func TestWorkflowOutputs(t *testing.T) {
+	wf := unmarshalWF(`
+metadata:
+  name: my-wf
+  namespace: my-ns
+spec:
+  entrypoint: main
+  templates:
+  - name: main
+    dag:
+      tasks:
+      - name: step-1
+        template: child
+  - name: child
+    container:
+      image: my-image
+    outputs:
+      parameters:
+      - name: my-param
+        valueFrom: 
+          path: /my-path
+`)
+	cancel, controller := newController(wf)
+	defer cancel()
+	woc := newWorkflowOperationCtx(wf, controller)
+	woc.operate()
+
+	assert.Equal(t, wfv1.NodeRunning, woc.wf.Status.Phase)
+
+	podInterface := controller.kubeclientset.CoreV1().Pods("my-ns")
+	list, err := podInterface.List(metav1.ListOptions{})
+	assert.NoError(t, err)
+	// two items -> two pods
+	assert.Len(t, list.Items, 1)
+
+	// mark all pods good
+	for _, pod := range list.Items {
+		pod.Status.Phase = apiv1.PodSucceeded
```

Comment:
@sarabala1979 @simster7 can I highlight this test? I don't think other tests do this, but it is possible to 

1. Create a `woc`
2. `woc.operate()`
3. Set the state of the pods created - in this example I set the phase and outputs.
4. `woc.operate()`
5. assert on workflow state

This means you CAN write a unit test that is as powerful as a functional e2e test - but faster and more robust.

Reply:
Please the existing function for `makePodsPhase`

Reply:
as I though, `makePodsPhase` does not support outputs

I'm starting to think of something like this.

```golang

	cancel, f := newFixture()
	defer cancel()
	f.Create(`
apiVersion: v1alpha1
kind: Workflow
metadata:
  name: my-wf
  namespace: my-ns
spec:
  entrypoint: main
  templates:
  - name: main
    dag:
      tasks:
      - name: step-1
        template: child
  - name: child
    container:
      image: my-image
    outputs:
      parameters:
      - name: my-param
        valueFrom: 
          path: /my-path
`).
		Reconcile().
		Merge(`
kind: Pod
metadata:
  name: ...
  annotations:
    workflows.argoproj.io/outputs: "{...}"
`).
		Reconcile().
		Expect(func(wf *wfv1.Workflow) {
			assert.Equal(t, wfv1.NodeSucceeded, wf.Status.Phase)
		
		})

```